lib/mayaUsd/ufe/UsdUndoDuplicateCommand.cpp

```diff
@@ -72,6 +72,13 @@ void UsdUndoDuplicateCommand::primInfo(
     // has a numerical suffix, increment it, otherwise append "1" to it.
     auto dstName = uniqueName(childrenNames, srcPrim.GetName());
     usdDstPath = parent.GetPath().AppendChild(TfToken(dstName));
+
+    srcLayer = MayaUsdUtils::defPrimSpecLayer(srcPrim);
+    if (!srcLayer) {
+        std::string err
+            = TfStringPrintf("No prim found at %s", srcPrim.GetPath().GetString().c_str());
+        throw std::runtime_error(err.c_str());
+    }
 }
```

Comment:
Hey @fowlertADSK   There is no need to call `MayaUsdUtils::defPrimSpecLayer `. This routine can be removed all together later. It is currently called in some Ufe1.0 but I believe this can be cleaned up as well.
https://github.com/Autodesk/maya-usd/blob/91994c05e287ca22d522556eee3d82eb4c93ecc4/lib/mayaUsd/ufe/UsdHierarchy.cpp#L149

> What you're suggesting is to change this to always grab the current edit target layer? With the new restriction we want to enforce, I think those two should always be the same, but there is a comment in the next function down that says we chose to not just grab the edit target layer.

By calling applyCommandRestriction(), you are guaranteed to be the correct layer. That's the whole idea, that this restrict function puts you in a correct layer to perform your change.
This layer can be retrieved from the edit target

_layer = srcPrim.GetStage()->GetEditTarget().GetLayer();

> The other primInfo static is used by Scene Item Ops so can't be moved to the constructor.

The name primInfo is confusing and can be simplified. This is how I would imagine the constructor to be 

```
{
    ufe::applyCommandRestriction(srcPrim, "duplicate");

    _layer = _srcPrim.GetStage()->GetEditTarget().GetLayer();

    auto parentPrim = srcPrim.GetParent();
    auto dstName = uniqueChildName(parentPrim, srcPrim.GetName());
    _usdDstPath = parentPrim.GetPath().AppendChild(TfToken(dstName));
}
```
Similar to UsdUndoRenameCommand, UsdUndoDuplicateCommand can return the duplicate Scene item which can be returned by UsdSceneItemOps::duplicateItem()

```
UsdSceneItem::Ptr duplicatedItem() const;  { return _ufeDuplicateItem; }

Ufe::SceneItem::Ptr UsdSceneItemOps::duplicateItem()
{
    auto duplicateCmd = UsdUndoDuplicateCommand::create(prim(), fItem->path());
    duplicateCmd->execute();
    return duplicateCmd->duplicatedItem();
}

Ufe::Duplicate UsdSceneItemOps::duplicateItemCmd()
{
    auto duplicateCmd = UsdUndoDuplicateCommand::create(prim(), fItem->path());
    duplicateCmd->execute();
    return Ufe::Duplicate(duplicateCmd->duplicatedItem(), duplicateCmd);
}
```

I hope it makes sense.